pybossa/cache/site_stats.py

```diff
@@ -157,28 +186,26 @@ def get_top5_users_24_hours():
     return top5_users_24_hours
 
 
-def allow_all_time(func):
-    @wraps(func)
-    def wrapper(days=30):
-        if days == 'all':
-            days = 999999
-        return func(days=days)
-
-    return wrapper
-
-
 @memoize_with_l2_cache(timeout=ONE_WEEK, timeout_l2=TWO_WEEKS,
                        cache_group_keys=['number_of_created_jobs'])
 @allow_all_time
 def number_of_created_jobs(days=30):
     """Number of created jobs"""
+    return number_of_created_jobs_interval(interval=str(days) + "days")
+
+
+@memoize_with_l2_cache(timeout=ONE_WEEK, timeout_l2=TWO_WEEKS,
+                       cache_group_keys=['number_of_created_jobs_interval'])
+@allow_all_time_interval
+def number_of_created_jobs_interval(interval="all"):
```

Comment:
By calling ```number_of_created_jobs```, it will cache twice. How about removing ```number_of_created_jobs_interval```?

Reply:
We need both functions, as they are used in unique ways. See:

https://github.com/bloomberg/pybossa/blob/67fb125f5fcf86b063f2d5ef38d39f879eef0b03/pybossa/jobs.py#L1361
https://github.com/bloomberg/pybossa/blob/main/pybossa/jobs.py#L1369

However, We should remove caching on `number_of_created_jobs` so it will not cache twice. This way both usages can be maintained, but they share a single implementation and cache.

Reply:
fixed via https://github.com/bloomberg/pybossa/commit/6d52b5365fa9ec031726b575e32438fa694b6b71